### PR TITLE
community: Add PolygonLastQuote Tool and Toolkit

### PR DESCRIPTION
**Pull request by virattt**
**Description:** 
In this PR, I am adding a `PolygonLastQuote` Tool, which can be used to get the latest price quote for a given ticker / stock. 

Additionally, I've added a Polygon Toolkit, which we can use to encapsulate future tools that we build for Polygon.

**Twitter handle:** [@virattt](https://twitter.com/virattt)

### CODE REVIEW

File: docs/docs/integrations/tools/polygon.ipynb [hunk 1]
```diff
@@ -63,18 +63,18 @@
...
+     "data": {
+      "text/plain": [
+       "{'results': {'P': 185.86, 'S': 1, 'T': 'AAPL', 'X': 11, 'i': [604], 'p': 185.81, 'q': 106551669, 's': 2, 't': 1705098436014023700, 'x': 12, 'y': 1705098436014009300, 'z': 3}}"
+      ]
+     },
+     "execution_count": 1,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
```
Comment (virattt) [inline docs/docs/integrations/tools/polygon.ipynb:77]: changes due to `make format`

**Comment by virattt** [comment]
There was a lot of boilerplate that I had to set up.

Lmk if I overdid it

File: docs/docs/integrations/toolkits/polygon.ipynb [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+   "source": [
+    "from langchain.agents import AgentType, initialize_agent\n",
+    "from langchain_community.agent_toolkits.polygon.toolkit import PolygonToolkit\n",
+    "from langchain_community.utilities.polygon import PolygonAPIWrapper\n",
+    "from langchain_openai import OpenAI\n",
+    "\n",
+    "llm = OpenAI(temperature=0, openai_api_key=\"\")\n",
+    "polygon = PolygonAPIWrapper()\n",
+    "toolkit = PolygonToolkit.from_polygon_api_wrapper(polygon)\n",
+    "agent = initialize_agent(\n",
```
Comment (hwchase17) [inline docs/docs/integrations/toolkits/polygon.ipynb:40]: can we not use initialize agent, but rather match the notebook here: https://python.langchain.com/docs/integrations/toolkits/gmail

**Review by hwchase17 — COMMENTED** [review]
lets NOT update langchain at all - trying to not make langchain depend on community as much going forward

**Pull request merged**